command.go

```diff
@@ -185,7 +185,8 @@ func (c *Command) parseFlags(args Args) (*flag.FlagSet, error) {
 	}
 
 	err = parseIter(set, c, args.Tail())
-	if err != nil {
+	// Continue parsing flags on failure during shell completion
+	if err != nil && !shellComplete {
```

Comment:
This basically allows for the context to continue to be parsed. Malformed flags will still be ignored and not part of the `set`, which is what v2 pre-release builds did.

Reply:
⭐⭐⭐ _(must change)_ @unRob can you add in here a detailed description of what `parseIter`, and why it's ok to skip errors from it when doing shell completion?

FYI cc @rliebz

Reply:
OH! Actually! I have a better idea: can you change this code to pass `shellComplete` into the `parseIter` function, ignore the errors inside the function itself, and add the descriptions I just requested inside the function's docstring?

Reply:
Great idea! 

- Changed the signature of `parseIter` to accept a fourth parameter `shellComplete bool` to deal with nil errors within.
- Decided to keep the `shellComplete` name, played with `ignoreErrors` instead, but it felt like a misnomer since `set.Parse` won't actually error on it.
- Modified a couple of consumers to pass `shellComplete` from their own context as well.
- Added a couple of lines to the docstring, hope my wording conveys the use for passing that argument.